Задание на животных и ООП

SlavaSTiwari opened the pull request:
Загрузка решения задание по изготовлению животных(кошка, собака, кенгуру) объектов и обучению ООП.

SlavaSTiwari committed:
Задание на животных и ООП

Загрузка решения задание по изготовлению животных(кошка, собака, кенгуру) объектов и обучению ООП.

SorokinIA commented on Animals.cpp:
```diff
@@ -1,61 +1,197 @@
+    public:
+        // Constructors:
+        Animal(){
+            name = "no name";
```
Это ок однако 
1) можно в конструкторе с параметрами выставить параметры по умолчанию
2) можно заполнять параметры класса через :

SorokinIA commented on Animals.cpp:
```diff
@@ -1,61 +1,197 @@
+class Cat: public Animal
+{
+    public:
+        void say(){
```
Рекомендую после объявления функции писать слово override иначе можно поспать в очень неприятную ситуацию

SorokinIA commented on Animals.cpp:
```diff
@@ -1,61 +1,197 @@
+    int c = 0;
+
+    cout << "\nThe 5 fattest cats are:" << endl;
+    for(int i=vect.size(); i>0; i--) {
```
А почему тут будут самое толстые коты?

SorokinIA commented on Animals.cpp:
```diff
@@ -1,61 +1,197 @@
+
+    // 3. Using sort function to find the fattest Animal:
+    std::sort(a1.begin(), a1.end(), [](Animal* a, Animal* b) {
+        return a->get_weight() < b->get_weight();
```
Почему для котов нельзя сразу компаратор было написать правильный если кот слева то...

SlavaSTiwari committed:
Задача на животных (ИСПРАВЛЕНО)

Добавлено:
--Ключевое слово "override" в нужных местах.
--Аргументы по умолчанию для конструктора класса "Animal".

Удалено:
--Библиотека booster и теперь используется <string>

SlavaSTiwari committed:
Merge branch 'main' into Self_work_tasks
